client/src/main/java/org/eclipse/hono/client/impl/HonoClientImpl.java

```diff
@@ -166,7 +168,13 @@ public HonoClient connect(
                         } else {
                             setConnection(conAttempt.result());
                             setContext(Vertx.currentContext());
-                            connectionHandler.handle(Future.succeededFuture(this));
+                            if (shutdown) {
+                                // if client was already shutdown in the meantime we give our best to cleanup connection
```

Comment:
why would want the *connection* and *context* being set in this case?

Reply:
Otherwise connection gets not closed in shutdownConnection and we get npe in shutdownConnection

Reply:
Ah, I see.